lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -171,48 +171,60 @@ public YarnSubmissionHelper setPreserveEvaluators(final boolean preserveEvaluato
                 " will set it to true.",
             YarnTypes.MIN_VERSION_KEEP_CONTAINERS_AVAILABLE);
 
-        applicationSubmissionContext.setKeepContainersAcrossApplicationAttempts(true);
+        this.applicationSubmissionContext.setKeepContainersAcrossApplicationAttempts(true);
       } else {
         LOG.log(Level.WARNING,
             "Hadoop version does not yet support KeepContainersAcrossApplicationAttempts. Driver restarts " +
                 "will not support recovering evaluators.");
 
-        applicationSubmissionContext.setKeepContainersAcrossApplicationAttempts(false);
+        this.applicationSubmissionContext.setKeepContainersAcrossApplicationAttempts(false);
       }
     } else {
-      applicationSubmissionContext.setKeepContainersAcrossApplicationAttempts(false);
+      this.applicationSubmissionContext.setKeepContainersAcrossApplicationAttempts(false);
     }
 
     return this;
   }
 
   /**
    * Sets the maximum application attempts for the application.
-   * @param maxApplicationAttempts
-   * @return
+   * @param maxApplicationAttempts max number of attempts to run the application.
+   * @return reference to self for chain calls.
    */
   public YarnSubmissionHelper setMaxApplicationAttempts(final int maxApplicationAttempts) {
-    applicationSubmissionContext.setMaxAppAttempts(maxApplicationAttempts);
+    this.applicationSubmissionContext.setMaxAppAttempts(maxApplicationAttempts);
     return this;
   }
 
   /**
    * Assign this job submission to a queue.
-   * @param queueName
-   * @return
+   * @param queueName name of the YARN job queue.
+   * @return reference to self for chain calls.
    */
   public YarnSubmissionHelper setQueue(final String queueName) {
     this.applicationSubmissionContext.setQueue(queueName);
     return this;
   }
 
   /**
+   * Specify whether the application manager should be run in unmanaged mode.
+   * If true, the Driver will run locally but still use YARN to allocate Evaluators.
+   * By default this is false, i.e. REEF Driver runs in its own container managed by YARN.
+   * @param isUnmanaged boolean flag whether to run Driver in unmanaged mode.
+   * @return reference to self for chain calls.
+   */
+  public YarnSubmissionHelper setUnmanagedAM(final boolean isUnmanaged) {
```

Comment:
Do we need the parameter? As in, can't `setUnmanagedAM` just set the Driver to be unmanaged?